atc/atccmd/command.go

```diff
@@ -96,387 +99,252 @@ const algorithmLimitRows = 100
 
-	DisplayUserIdPerConnector map[string]string `long:"display-user-id-per-connector" description:"Define how to display user ID for each authentication connector. Format is <connector>:<fieldname>. Valid field names are user_id, name, username and email, where name maps to claims field username, and username maps to claims field preferred username"`
-}
-
-type Migration struct {
-	lockFactory lock.LockFactory
-
-	Postgres               flag.PostgresConfig `group:"PostgreSQL Configuration" namespace:"postgres"`
-	EncryptionKey          flag.Cipher         `long:"encryption-key"     description:"A 16 or 32 length key used to encrypt sensitive information before storing it in the database."`
-	OldEncryptionKey       flag.Cipher         `long:"old-encryption-key" description:"Encryption key previously used for encrypting sensitive information. If provided without a new key, data is decrypted. If provided with a new key, data is re-encrypted."`
-	CurrentDBVersion       bool                `long:"current-db-version" description:"Print the current database version and exit"`
-	SupportedDBVersion     bool                `long:"supported-db-version" description:"Print the max supported database version and exit"`
-	MigrateDBToVersion     int                 `long:"migrate-db-to-version" description:"Migrate to the specified database version and exit"`
-	MigrateToLatestVersion bool                `long:"migrate-to-latest-version" description:"Migrate to the latest migration version and exit"`
-}
+	DefaultCpuLimit      *int          `yaml:"default_task_cpu_limit,omitempty"`
+	DefaultMemoryLimit   *string       `yaml:"default_task_memory_limit,omitempty"`
+	InterceptIdleTimeout time.Duration `yaml:"intercept_idle_timeout,omitempty"`
 
-func (m *Migration) Execute(args []string) error {
-	lockConn, err := constructLockConn(defaultDriverName, m.Postgres.ConnectionString())
-	if err != nil {
-		return err
-	}
-	defer lockConn.Close()
-
-	m.lockFactory = lock.NewLockFactory(lockConn, metric.LogLockAcquired, metric.LogLockReleased)
-
-	if m.MigrateToLatestVersion {
-		return m.migrateToLatestVersion()
-	}
-	if m.CurrentDBVersion {
-		return m.currentDBVersion()
-	}
-	if m.SupportedDBVersion {
-		return m.supportedDBVersion()
-	}
-	if m.MigrateDBToVersion > 0 {
-		return m.migrateDBToVersion()
-	}
-	if m.OldEncryptionKey.AEAD != nil {
-		return m.rotateEncryptionKey()
-	}
-	return errors.New("must specify one of `--migrate-to-latest-version`, `--current-db-version`, `--supported-db-version`, `--migrate-db-to-version`, or `--old-encryption-key`")
-}
+	CLIArtifactsDir flag.Dir `yaml:"cli_artifacts_dir,omitempty"`
+	WebPublicDir    flag.Dir `yaml:"web_public_dir"`
 
-func (cmd *Migration) currentDBVersion() error {
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		nil,
-		nil,
-	)
+	BaseResourceTypeDefaults flag.File `yaml:"base_resource_type_defaults,omitempty"`
```

Comment:
Rather than pointing this to a file, now that we have an actual config file I think it'd make sense to just have it inline.

Reply:
Oh like have it be inline within the config file but there is a flag that still accepts the `base_resource_type_defaults` as a file?